source/docs/software/commandbased/organizing-command-based.rst

```diff
@@ -0,0 +1,299 @@
+
+Instance Command Factory Methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A command like the intake-running command is conceptually related to exactly one subsystem: the ``Intake``. As such, it makes sense to put a ``commandRun`` method as an instance method of the ``Intake`` class:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    public class Intake {
+        // [code for motor controllers, configuration, etc.]
+        // ...
+
+        public Command commandRun() {
+            return new StartEndCommand(() -> this.set(1), () -> this.set(0), this);
+        }
+    }
+
+  .. code-tab:: c++
+
+    // TODO
+
+Notice how since we are in the ``Intake`` class, we no longer refer to ``intake``; instead, we use the ``this`` keyword to refer to the current instance.
+
+.. warning:: To preserve encapsulation, avoid referring to ``private`` variables and methods of the ``Intake`` class from within the ``commandRun`` method, even though they are technically accessible. (For instance, the ``commandRun`` method should use the subsystem's public ``set`` method instead of interfacing directly with the motor controllers.)
+
+Using this new factory method in command groups and button bindings is highly expressive:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    intakeButton.whileHeld(intake.commandRun());
+
+    Command intakeAndShoot = intake.commandRun().alongWith(new RunShooter());
+
+    Command autonomousCommand = new SequentialCommandGroup(
+        intake.commandRun().withTimeout(5),
+        new WaitCommand(3),
+        intake.commandRun().withTimeout(5)
+    );
+
+  .. code-tab:: c++
+
+    // TODO
+
+Adding a parameter to the ``commandRun`` method to provide the exact percentage to run the intake is easy and allows for even more flexibility.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    public Command commandRun(double percent) {
+        return new StartEndCommand(() -> this.set(percent), () -> this.set(0), this);
+    }
+
+  .. code-tab:: c++
+
+    // TODO
+
+For instance, this code creates a command group that runs the intake forwards for two seconds, waits for two seconds, and then runs the intake backwards for five seconds.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    Command intakeRunSequence = intake.commandRun(1).withTimeout(2)
+        .andThen(new WaitCommand(2))
+        .andThen(intake.commandRun(-1).withTimeout(5));
+
+  .. code-tab:: c++
+
+    // TODO
+
+Static Command Factories
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The factory methods presented above do have some negatives. They clutter up the `Intake` class (and the namespace of `Intake` methods) with command-related methods. With that pattern, it's also possible to break the intended encapsulation of the subsystem (as the warning box alludes to), since code in these factory methods can access `private` members of the `Intake` class. To avoid these pitfalls, it's possible to define intake-related methods as static members of a separate `IntakeCommands` class:
```

Comment:
What do you see duplicated?

> Regardless, I see static factories more for multi-subsystem commands than for single-subsystem ones.

I personally agree, but I can see some teams being more comfortable with "all commands are static factories" because (one possible reason) that's just fewer bits of information they have to drill into the heads of a bunch of novices. (Maybe I should add that to the "pros" list for static single-subsystem factories.)

Reply:
> What do you see duplicated?

There's a static command factory section and a static command group factory section.

> (Maybe I should add that to the "pros" list for static single-subsystem factories.)

I'd be very careful with wording that pro, if at all. I'm not sure knowing to only do one thing is good.